src/coreclr/src/vm/exceptionhandling.h

```diff
@@ -13,10 +13,6 @@
 #include "eexcp.h"
 #include "exstatecommon.h"
 
-#if defined(TARGET_ARM) || defined(TARGET_X86)
-#define USE_PER_FRAME_PINVOKE_INIT
```

Comment:
This will need matching changes in the JIT as well. You should see a functional test failures without matching changes in the JIT.

Reply:
The JIT today links/unlinks the ICF in the method prolog/epilog, and activates/deactivates the ICF around the pinvoke call. It does this for both 32-bits and 64-bits. The fix here is to match the behavior in the EH codebase.

As a second step after this fix is in, I can look into merging the linking+activation in one step around the pinvoke call (this will match the behavior of R2R code as well), but that would be a slightly bigger change I believe